plugin/storage/es/samplingstore/storage.go

```diff
@@ -0,0 +1,221 @@
+	val := dbmodel.ProbabilitiesAndQPS{
+		Hostname:      hostname,
+		Probabilities: probabilities,
+		QPS:           qps,
+	}
+	s.writeProbabilitiesAndQPS(writeIndexName, ts, val)
+	return nil
+}
+
+func (s *SamplingStore) GetLatestProbabilities() (model.ServiceOperationProbabilities, error) {
+	ctx := context.Background()
+	clientFn := s.client()
+	indices, err := getLatestIndices(s.samplingIndexPrefix, s.indexDateLayout, clientFn, s.samplingIndexRolloverFrequency, s.adaptiveSamplingLookback)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get latest indices: %w", err)
+	}
+	searchResult, err := clientFn.Search(indices...).
+		Size(s.maxDocCount).
+		IgnoreUnavailable(true).
+		Do(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to search for Latest Probabilities: %w", err)
+	}
+	lengthOfSearchResult := len(searchResult.Hits.Hits)
+	if lengthOfSearchResult == 0 {
+		return nil, nil
+	}
+
+	var latestProbabilities dbmodel.TimeProbabilitiesAndQPS
+	latestTime := time.Time{}
+	for _, hit := range searchResult.Hits.Hits {
+		var data dbmodel.TimeProbabilitiesAndQPS
+		if err = json.Unmarshal(*hit.Source, &data); err != nil {
+			return nil, fmt.Errorf("unmarshalling documents failed: %w", err)
+		}
+		if data.Timestamp.After(latestTime) {
+			latestTime = data.Timestamp
+			latestProbabilities = data
+		}
+	}
+	return latestProbabilities.ProbabilitiesAndQPS.Probabilities, nil
+}
+
+func (s *SamplingStore) writeThroughput(indexName string, ts time.Time, throughputs []*model.Throughput) {
+	s.client().Index().Index(indexName).Type(throughputType).
+		BodyJson(&dbmodel.TimeThroughput{
+			Timestamp:  ts,
+			Throughput: dbmodel.FromThroughputs(throughputs),
+		}).Add()
+}
+
+func (s *SamplingStore) writeProbabilitiesAndQPS(indexName string, ts time.Time, pandqps dbmodel.ProbabilitiesAndQPS) {
+	s.client().Index().Index(indexName).Type(probabilitiesType).
+		BodyJson(&dbmodel.TimeProbabilitiesAndQPS{
+			Timestamp:           ts,
+			ProbabilitiesAndQPS: pandqps,
+		}).Add()
+}
+
+func (s *SamplingStore) CreateTemplates(samplingTemplate string) error {
+	_, err := s.client().CreateTemplate("jaeger-sampling").Body(samplingTemplate).Do(context.Background())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getLatestIndices(indexPrefix, indexDateLayout string, clientFn es.Client, rollover time.Duration, maxDuration time.Duration) ([]string, error) {
+	ctx := context.Background()
+	now := time.Now().UTC()
+	end := now.Add(-maxDuration)
+	endIndex := indexWithDate(indexPrefix, indexDateLayout, end)
+	for {
+		currentIndex := indexWithDate(indexPrefix, indexDateLayout, now)
+		exists, err := clientFn.IndexExists(currentIndex).Do(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to check index existence: %w", err)
+		}
+		if exists {
+			return []string{currentIndex}, nil
+		}
+		if currentIndex == endIndex {
+			return nil, fmt.Errorf("falied to find latest index")
+		}
+		now = now.Add(rollover)
```

Comment:
I don't know why rollover is negative (bad coding pattern)
```suggestion
		now = now.Add(rollover) // rollover is negative
```